core/dbt/parser/models.py

```diff
@@ -137,6 +152,13 @@ def parse_python_model(self, node, config, context):
             msg = validator_error_message(exc)
             raise ParsingException(msg, node=node) from exc
 
+        # We are doing a validator and a parser because visit_FunctionDef in parser
+        # would actually make the parser not doing the visit_Calls any more
+        dbtValidator = PythonValidationVisitor()
+        dbtValidator.visit(tree)
+        if len(dbtValidator.dbt_errors) != 0:
+            raise ParsingException("\n".join(dbtValidator.dbt_errors), node=node)
```

Comment:
This Exception will result in a chain of error message instead of just one message that give dbt_errors, Do we want to do something to make the error message cleaner?

Reply:
Or maybe I have something in the target branch that uncommented some error handling, will look into it tomorrow

Reply:
Is the goal here to raise a nicer error when dbt encounters invalid Python syntax? Here's what I see locally. It gets the job done, but it is pretty verbose! The filename also shows up as `"<unknown>"`
```
$ dbt parse
10:44:49  Running with dbt=1.3.0-a1
10:44:49  Start parsing.
10:44:49  Dependencies loaded
10:44:50  ManifestLoader created
Traceback (most recent call last):
  File "/Users/jerco/dev/product/dbt-core/core/dbt/parser/models.py", line 150, in parse_python_model
    tree = ast.parse(node.raw_sql)
  File "/usr/local/Cellar/python@3.9/3.9.12/Frameworks/Python.framework/Versions/3.9/lib/python3.9/ast.py", line 50, in parse
    return compile(source, filename, mode, flags,
  File "<unknown>", line 12
    as'difajsdlf
    ^
SyntaxError: invalid syntax

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/jerco/dev/product/dbt-core/core/dbt/main.py", line 129, in main
    results, succeeded = handle_and_check(args)
  File "/Users/jerco/dev/product/dbt-core/core/dbt/main.py", line 192, in handle_and_check
    task, res = run_from_args(parsed)
  File "/Users/jerco/dev/product/dbt-core/core/dbt/main.py", line 239, in run_from_args
    results = task.run()
  File "/Users/jerco/dev/product/dbt-core/core/dbt/task/parse.py", line 93, in run
    self.get_full_manifest()
  File "/Users/jerco/dev/product/dbt-core/core/dbt/task/parse.py", line 74, in get_full_manifest
    manifest = loader.load()
  File "/Users/jerco/dev/product/dbt-core/core/dbt/parser/manifest.py", line 349, in load
    self.parse_project(
  File "/Users/jerco/dev/product/dbt-core/core/dbt/parser/manifest.py", line 473, in parse_project
    parser.parse_file(block)
  File "/Users/jerco/dev/product/dbt-core/core/dbt/parser/base.py", line 392, in parse_file
    self.parse_node(file_block)
  File "/Users/jerco/dev/product/dbt-core/core/dbt/parser/base.py", line 366, in parse_node
    self.render_update(node, config)
  File "/Users/jerco/dev/product/dbt-core/core/dbt/parser/models.py", line 182, in render_update
    self.parse_python_model(node, config, context)
  File "/Users/jerco/dev/product/dbt-core/core/dbt/parser/models.py", line 153, in parse_python_model
    raise ParsingException(msg, node=node) from exc
dbt.exceptions.ParsingException: Parsing Error in model in_quarter_magic_factors (models/marts/marketing/in_quarter_magic/in_quarter_magic_factors.py)
  invalid syntax (<unknown>, line 12)
10:44:50  Encountered an error:
Parsing Error in model in_quarter_magic_factors (models/marts/marketing/in_quarter_magic/in_quarter_magic_factors.py)
  invalid syntax (<unknown>, line 12)
```

Reply:
Oh that's a different one I haven't got to. If you try the example in tests of this PR you will get the file path, but still verbose

Reply:
I think we won't be able to have a file name instead of "unknown" since we're parsing a string. The file name does show up in multiple places.

Reply:
Oh no worries, we can raise an error with the node info with has file names I just haven't got to this yet